technic/machines/HV/quarry.lua

```diff
@@ -305,6 +332,14 @@ local digiline_def = function(pos, _, channel, msg)
 		meta:set_int("size", size)
 		reset_quarry(pos)
 		set_quarry_status(pos)
+	elseif msg.command == "max_depth" then
+		local max_depth = tonumber(msg.value)
+		if not max_depth or max_depth < 0 or max_depth > quarry_max_depth or max_depth == meta:get_int("max_depth") then
+			return
+		end
+		meta:set_int("max_depth", max_depth)
```

Comment:
I mean here instead of returning if max_depth > quarry_max_depth do math.min(max_depth, quarry_max_depth)

Reply:
err, this doesn't really work: in case the digiline-set value is greater than the maximum it should just set it to max (or discard it, not to damage anything)
Is this your recommendation: ?
```lua
if max_depth > quarry_max_depth then
  math.min(max_depth, quarry_max_depth)
end
```

What is expected here for the cases:
* invalid number :arrow_right: discard
* negative or zero :arrow_right: set to maximum?
* above limit :arrow_right: set to current maximum setting

Reply:
Yes, I meant just remove from if, fall through and do meta:set_int("max_depth",math.min(max_depth, quarry_max_depth))

Reply:
its not that easy: the value could still be `nil` (crash on `math.min()`) or negative (invalid setting),
anyway: separated those checks and added a fallback to max if the value if above the limit.

Reply:
Sorry about confusion, I should have written it more clearly from beginning... There was already nil check and less than 0 check, I really meant to only remove `or max_depth > quarry_max_depth` part from if line but keep rest of it.